app/models/user.rb

```diff
@@ -153,10 +153,17 @@ def num_rated_mentored_solutions
   end
 
   def mentor_rating
-   @mentor_rating ||=
-      solution_mentorships.where.not(rating: nil).
-                           pluck(Arel.sql("AVG(rating) as r")).
-                           first.to_f.round(2)
+    return @mentor_rating if @mentor_rating
```

Comment:
So for memoizing, I prefer using the `||=` pattern with a block, rather than setting instance variables in multiple places. ie, if you use something like this, you can just remove the `@mentor_rating = ` bits from your code and it *should* just work :)

```
@mentor_rating ||= begin
...
end
```